blockchain/blockchain.go

```diff
@@ -31,15 +31,15 @@ var stateKey = []byte("BlockchainState")
 
 // Immutable Root of blockchain
 type Root interface {
-	// ChainID precomputed from GenesisDoc
-	ChainID() string
 	// GenesisHash precomputed from GenesisDoc
 	GenesisHash() []byte
 	GenesisDoc() genesis.GenesisDoc
 }
 
 // Immutable pointer to the current tip of the blockchain
 type Tip interface {
+	// ChainID precomputed from GenesisDoc
+	ChainID() string
```

Comment:
Curious why we're moving `ChainID` from root to tip; its static so not really sure why we are passing it around with each block. Perhaps better safety would be my assumption....?

Reply:
Though you grab tip immutably with `Tip()` so there is a copy when you pull a tip explicitly, but mostly we are just accessing the single instance embedded in `blockchain` - both when accessing values or doing a `Commit()`. Having said that I do agree with you that it seems more logical in `Root` as a static value. The reason I moved it is with the addition of `ChainID` `Transactor` need only ask for a `Tip` rather than a whole `Blockchain`. I could have just added `ChainID` as an argument to `Transactor`. Let me think about that, maybe that is a nicer way to go... A counter-argument is if you want `Tip` to be the chain information you need to sign at a particular height, then it seems like `ChainID` belongs with other tip data.

Reply:
Thanks. Realized 75% thru the review why.... Should have removed the comment. But always good to learn!